src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/DependencyGraphSpecRequestProvider.cs

```diff
@@ -33,14 +34,14 @@ public class DependencyGraphSpecRequestProvider : IPreLoadedRestoreRequestProvid
             _providerCache = providerCache;
         }
 
-        public Task<IReadOnlyList<RestoreSummaryRequest>> CreateRequests(RestoreArgs restoreContext)
+        public Task<IReadOnlyCollection<RestoreSummaryRequest>> CreateRequests(RestoreArgs restoreContext)
```

Comment:
This is a breaking change on a public method on a public class. I find it very, very difficult to imagine why anyone would call this API, but what's the performance impact if you keep the same datatype and copy the results into a list after the `Parallel.ForEach` completes?

Reply:
Thanks for the concern.  I did more digging and found that `ToList<T>()` would be slow but `ToArray()` is fast and gets rid of the need to change the return type.  I've updated the PR!